File: src/main/java/red/tetracube/iot/smartigloohub/data/entities/Account.java
package red.tetracube.iot.smartigloohub.data.entities;

import red.tetracube.iot.smartigloohub.enumerations.AccountHubRole;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Table(schema = "igloo_hub", name = "accounts")
public class Account {

    @Id
    private UUID id;

    @Column(name = "username", nullable = false, unique = true)
    private String username;

    @Column(name = "password", nullable = false)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(name = "role", nullable = false)
    private AccountHubRole role;

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public AccountHubRole getRole() {
        return role;
    }
}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/data/entities/Account.kt
package red.tetracube.iot.smartigloohub.data.entities

import java.util.UUID
import javax.persistence.*

@Entity
@Table(schema = "igloo_hub", name = "accounts")
data class Account(
    @Id
    val id: UUID? = null,

    @Column(name = "username", nullable = false, unique = true)
    val username: String? = null,

    @Column(name = "password", nullable = false)
    val password: String? = null,

    @OneToMany(fetch = FetchType.LAZY, targetEntity = HouseAccount::class, mappedBy = "account")
    val housesAccounts: List<HouseAccount> = emptyList()
)
